task-sdk/src/airflow/sdk/definitions/dag.py

```diff
@@ -1259,7 +1261,10 @@ def _run_task(*, ti, run_triggerer=False):
                 ti.next_kwargs = {"event": event.payload} if event else msg.next_kwargs
                 log.info("[DAG TEST] Trigger completed")
 
-                ti.set_state(State.SUCCESS)
```

Comment:
Can we not do `ti.set_state(State.SCHEDULED)`? Trying to avoid session imports in sdk code

Reply:
Unfortunately not, since set_state calls refresh_from_db which resets next_kwargs.

But session is needed right now anyway for dag.test and task.test